app/main.py

```diff
@@ -2,24 +2,63 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, None)
```

Comment:
Instead of setting the attribute to `None` when the value is out of range, it would be more appropriate to raise a `ValueError` with a descriptive message. This will make it clear that the value is invalid and why.